release/src/milestones.ts

```diff
@@ -54,13 +54,17 @@ async function getOriginalPR({
     pull_number: pullRequestNumber,
   });
 
-  if (pull?.data && isBackport(pull.data)) {
-    return getOriginalPR({
-      github,
-      repo,
-      owner,
-      pullRequestNumber: pull.data.number
-    });
+  if (pull?.data && isBackport(pull.data) && pull.data.body) {
+    const sourcePRNumber = getBackportSourcePRNumber(pull.data.body);
+    if (sourcePRNumber && sourcePRNumber !== pullRequestNumber) {
```

Comment:
here was the missing logic - we need to parse the backport PR body for the name of the PR it backports, we also need to check to make sure it's not the same as the currently parsed PR to prevent infinite recursion.